instrumentation/apache-httpasyncclient-4.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpasyncclient/ApacheHttpAsyncClientSingletons.java

```diff
@@ -24,33 +15,10 @@ public final class ApacheHttpAsyncClientSingletons {
   private static final Instrumenter<ApacheHttpClientRequest, HttpResponse> INSTRUMENTER;
 
   static {
-    ApacheHttpAsyncClientHttpAttributesGetter httpAttributesGetter =
-        new ApacheHttpAsyncClientHttpAttributesGetter();
-
-    InstrumenterBuilder<ApacheHttpClientRequest, HttpResponse> builder =
-        Instrumenter.<ApacheHttpClientRequest, HttpResponse>builder(
-                GlobalOpenTelemetry.get(),
-                INSTRUMENTATION_NAME,
-                HttpSpanNameExtractor.builder(httpAttributesGetter)
-                    .setKnownMethods(CommonConfig.get().getKnownHttpRequestMethods())
-                    .build())
-            .setSpanStatusExtractor(HttpSpanStatusExtractor.create(httpAttributesGetter))
-            .addAttributesExtractor(
-                HttpClientAttributesExtractor.builder(httpAttributesGetter)
-                    .setCapturedRequestHeaders(CommonConfig.get().getClientRequestHeaders())
-                    .setCapturedResponseHeaders(CommonConfig.get().getClientResponseHeaders())
-                    .setKnownMethods(CommonConfig.get().getKnownHttpRequestMethods())
-                    .build())
-            .addAttributesExtractor(
-                HttpClientPeerServiceAttributesExtractor.create(
-                    httpAttributesGetter, CommonConfig.get().getPeerServiceResolver()))
-            .addOperationMetrics(HttpClientMetrics.get());
-    if (CommonConfig.get().shouldEmitExperimentalHttpClientTelemetry()) {
-      builder
-          .addAttributesExtractor(HttpExperimentalAttributesExtractor.create(httpAttributesGetter))
-          .addOperationMetrics(HttpClientExperimentalMetrics.get());
-    }
-    INSTRUMENTER = builder.buildClientInstrumenter(HttpHeaderSetter.INSTANCE);
+    INSTRUMENTER =
+        HttpClientInstrumenterFactory.builder(new ApacheHttpAsyncClientHttpAttributesGetter())
+            .instrumenterBuilder(INSTRUMENTATION_NAME)
```

Comment:
does it make sense to pass `INSTRUMENTATION_NAME` directly into `.builder()` method on the line above?